ENH: Support line-of-sight fields for particle types (#5001)
yt-project/yt: doc/source/analyzing/fields.rst

    # Set to a three-vector for an off-axis component
    dd.set_field_parameter("axis", [0.3, 0.4, -0.7])
    print(dd["gas", "velocity_los"])
    # particle fields are supported too!
    print(dd["all", "particle_velocity_los"])

.. warning::

    If you need to change the axis of the line of sight on the *same* data container
    (sphere, box, cylinder, or whatever), you will need to delete the field using (e.g.)
    ``del dd['velocity_los']`` and re-generate it.

At this time, this functionality is enabled for the velocity and magnetic vector
fields, ``('gas', 'velocity_los')`` and ``('gas', 'magnetic_field_los')`` for

> the ``"gas"`` field type, and it is also enabled for every particle type with

Reviewer: concision++
```suggestion
the ``"gas"`` field type, as well as every particle type with
```
